lib/OpenQA/Needles.pm

```diff
@@ -0,0 +1,63 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Needles;
+use Mojo::Base -strict, -signatures;
+
+use Exporter qw(import);
+use File::Basename;
+use File::Spec;
+use File::Spec::Functions qw(catdir);
+use OpenQA::App;
+use OpenQA::Git;
+use OpenQA::Log qw(log_error);
+use OpenQA::Utils qw(prjdir sharedir);
+use Mojo::File qw(path);
+
+our @EXPORT = qw(temp_dir is_in_temp_dir needle_temp_dir locate_needle _locate_needle_for_ref);
```

Comment:
I prefer to explicitly export/import functions in most cases.
```suggestion
our @EXPORT_OK = qw(temp_dir is_in_temp_dir needle_temp_dir locate_needle _locate_needle_for_ref);
```
Also it's very uncommon to export a function starting with an underscore. Such functions are usually only used inside of a package and seen as "private".